tests/test_20_use_case_topology.py

```diff
@@ -229,5 +229,114 @@ def test_011_intra_domain_link_down_path_found(self):
         assert len(initial_topology["nodes"]) == len(updated_topology["nodes"]), "Number of nodes changed"
         assert len(initial_topology["links"]) == len(updated_topology["links"]), "Number of links changed"
 
+    @pytest.mark.xfail(reason="The L2VPN status remains up after changing the status of an associated node to down")
+    def test_020_node_down(self):
+        """
+        Test Use Case 4: Node Status Change (Down)
+        
+        Verify that when a node goes down:
+        1. SDX Controller updates the status of the node, its ports, and links
+        2. SDX Controller updates affected L2VPN statuses to "error"
+        3. SDX Controller refuses to provision new services on the node
+        """
+        def set_node(node, status, target):
+            if status == 'down':
+                node.cmd(f"ovs-vsctl set-controller {node.name} {target}")
+                node.cmd(f"ovs-vsctl get-controller {node.name}") 
+            else:
+                node.cmd(f"ovs-vsctl set-controller {node.name} {target}")
+                node.cmd(f"ovs-vsctl get-controller {node.name}")
```

Comment:
Gretel, can you please move this function to the helper class? this. can be used by other tests so having it defined there is better we can reuse it.